### PR TITLE
feat(layout): adds a menu bar item to reset the current layout

### LINKED ISSUE #492: [Layout] Reset layout to default
**Is your feature request related to a problem? Please describe.**
If I screw up my layout I'd like an easy way to reset this.

**Describe the solution you'd like**
An item in the menu bar would be great.

**Describe alternatives you've considered**
Opening the dev console and clearing local storage items.

**Additional context**
The layout should not affect modV's current output. If I'm at a show I should be able to reset the layout with no interruptions to the visuals.

### PR DESCRIPTION
**Pull request by 2xAA**
This refactors the existing creation error code to allow for a solution to reset the layout without
losing your current project.

closes #492

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -338,14 +339,27 @@ export default {
...
+    },
+
+    /**
+     * @description Restarts Golden Layout.
+     * We increment a variable which is assigned to the key of the root Golden Layout element.
+     * If the key is updated, the element is forced to dismount and mount again.
+     */
+    restartLayout() {
+      this.resetGoldenLayoutState();
+      this.triggerUiRestart++;
```
Comment (TimPietrusky) [inline src/App.vue:362]: Why is this needed?

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
I have one question related to the code and one questoin regarding this warning that comes up when you reset the layout:

```Did not remove inputLink. Could not find input with id undefined```. 

Why is `removeInputLink` called at all when we do this? Is the layout somehow also connected to an `inputLink`?

File: src/App.vue (hunk 1 above)
Comment (2xAA) [reply src/App.vue:362]: > I have one question related to the code and one questoin regarding this warning that comes up when you reset the layout:
> 
> `Did not remove inputLink. Could not find input with id undefined`.
> 
> Why is `removeInputLink` called at all when we do this? Is the layout somehow also connected to an `inputLink`?

Yes, the layout contains the Gallery. Each Gallery Item is an active Module, so when the layout resets the gallery must unmount the Gallery Items and also clear up the active Modules, which does a check for InputLinks. As the Gallery Item active Modules are special Gallery ones, they have no inputs and we get these warnings.

We should add a case to check if the active Module is a module in the gallery and suppress these warnings.

Comment (2xAA) [reply src/App.vue:362]: >Why is this needed?

In Vue we can't programmatically unmount a component without using a `v-if` which unfortunately removes the elements from the DOM. We can force the component to re-mount without changing the DOM using `:key`, something we need to do to make Golden Layout initialise again with the updated layout state.

**Review by TimPietrusky — APPROVED** [review]
LGTM
